airflow/providers/amazon/aws/links/emr.py

```diff
@@ -66,3 +82,98 @@ def get_log_uri(
         return None
     log_uri = S3Hook.parse_s3_url(cluster_info["LogUri"])
     return "/".join(log_uri)
+
+
+class EmrServerlessLogsLink(BaseAwsLink):
+    """Helper class for constructing Amazon EMR Serverless Logs Link."""
+
+    name = "Spark Driver stdout"
+    key = "emr_serverless_logs"
+
+    def get_link(
+        self,
+        operator: BaseOperator,
+        *,
+        ti_key: TaskInstanceKey,
+    ) -> str:
+        """
+        Link to Amazon Web Services Console.
+
+        :param operator: airflow operator
+        :param ti_key: TaskInstance ID to return link for
+        :return: link to external system
+        """
+        conf = XCom.get_value(key=self.key, ti_key=ti_key)
+        if not conf:
+            return ""
+        hook = EmrServerlessHook(aws_conn_id=conf.get("conn_id"))
+        resp = hook.conn.get_dashboard_for_job_run(
+            applicationId=conf.get("application_id"), jobRunId=conf.get("job_run_id")
+        )
+        o = urlparse(resp["url"])
+        return o._replace(path="/logs/SPARK_DRIVER/stdout.gz").geturl()
```

Comment:
Is there a way to make this more robust? There's unfortunately not a way to store the URL, and this was definitely part of the challenge in implementing this - the link generated by `get_dashboard_for_job_run` is an ephemeral, one-time pre-signed link. So it expires an hour after it generates _or_ after being used, whichever comes first.

Reply:
Have you UAT tested what happens when you trigger this link with a web server that has no internet access @dacort?

I originally imagined this with more client side UI scripting. I.e. having a new button in the task details which when clicked kicks off the above code, that way the link would only be fetched if someone intentionally wanted it. But I think this should be fine too as long as it fails gracefully if no internet access.

Reply:
I'll double-check. As implemented today, the link only generates if somebody clicks the "Details" tab of the specific Task in question. While not as granular as I would like, I think it's still acceptable.

Reply:
With no internet access, the API call to `extra_links` (for the links that require outbound access) results in an HTTP 500. The way this renders in the Airflow UI is as below with grayed out buttons.
<img width="549" alt="image" src="https://github.com/apache/airflow/assets/1512/647a99fd-4050-4959-9514-fcf1af8e3ef3">

Note that this takes a few seconds to fail and show up - None of the extra link buttons show until all of the calls have finished.

Reply:
> Note that this takes a few seconds to fail and show up - None of the extra link buttons show until all of the calls have finished.

Is the rest of the Details dialogue usable during this time? Or is the whole thing wedged for that time? Also how long is a "few seconds" more precisely? Like 1-3s or 3+s?

Reply:
Yes, the rest of the Details dialogue is still usable. It's just the "Extra Links" section that doesn't populate (see below for before and after).

Unfortunately it takes about 8-12s for the call to `extra_links` to timeout. In testing, the first failed call usually fails around 7-8s and then second fails after 10-12s. 

Before:
<img width="672" alt="image" src="https://github.com/apache/airflow/assets/1512/937851b2-c7e0-4357-ab3d-84fe041ffb46">

After: 
<img width="660" alt="image" src="https://github.com/apache/airflow/assets/1512/8b12dc0f-d0d8-43e6-9644-7875c9294a30">

Reply:
> Unfortunately it takes about 8-12s for the call to extra_links to timeout.

Oh boy, that's quite a long time! Should we put some shorter timeouts in place so that it doesn't block the generation of the other links for so long?

Reply:
It is, especially given _none_ of the links show up until _all_ of the requests have finished. Some shorter timeouts would likely be good. Another option is handling specific errors (like host not found or connection failed)...but I'd rather have shorter timeouts than trying to catch specific things.

Reply:
That sounds like a reasonable approach. Do you mind having a go at that? I don't think I'd like to merge this with 10+ second latency for the link loading in failure cases.

Reply:
Yep, will look into it!

Reply:
OK, cool, what I did here was change the `total_max_attempts` to "1" specifically for the hook that calls `get_dashboard_for_job_run`. With a default timeout of 60s, this means that the calls will fail fast. I'm fine with this approach, even if they fail for a network blip.